src/ui/windows/TogglDesktop/TogglDesktop/ui/views/Timer.xaml.cs

```diff
@@ -27,64 +26,16 @@ public partial class Timer
         public Timer()
         {
             this.InitializeComponent();
-
-            this.setupSecondsTimer();
+            ViewModel = new TimerViewModel(false);
+            ViewModel.WhenValueChanged(x => x.IsRunning).Subscribe(_ => StartStopClick?.Invoke(this, EventArgs.Empty));

```

Comment:
`IsRunning` is changed != start/stop was clicked. `IsRunning` can be changed from the outside, so this is a logical error.

Reply:
@atsvirchkova this one is still not addressed

Reply:
Well, yes. The only difference between the `StartStopClicked` and `IsRunning` changing is that `IsRunning` can be set using keyboard. But the only thing that was done on `StartStopClicked` is calling `closeEditPopup()`. Is it intended to work differently in these two cases (I mean the keyboard case and the mouse case)?

Reply:
Not only the keyboard. Any update coming from the library can cause `IsRunning` to change. It seems like `TimerViewModel` doesn't apply those updates, which is causing the bug https://github.com/toggl-open-source/toggldesktop/pull/4170#pullrequestreview-435640135

Reply:
it seems like it's fine to close the edit popup every time when `IsRunning` is changed though